webapp/src/pages/PlayPage/PlayPage.tsx

```diff
@@ -1,42 +1,65 @@
+import './PlayPage.css';
 import * as React from 'react';
-import { ButtonVariant, IconVariant } from '../../shared/components';
+import {
+  ButtonProps,
+  ButtonVariant,
+  IconVariant,
+  Input,
+  InputVariant,
+  MediumAvatar,
+  NavigationBar,
+  ReactiveButton,
+  RowItem,
+  RowsList,
+} from '../../shared/components';
 import { useNavigate } from 'react-router-dom';
-import { MainTabTemplate } from '../../shared/components/index';
-import { SearchContextProvider } from '../../shared/context/SearchContext';
-import { PLAY_GAME_TRAIN_URL } from '../../shared/urls';
-import { ENTRIES_LIMIT } from '../../shared/constants';
-import { Query } from '../../shared/types';
+import {
+  AVATAR_EP_URL,
+  PLAY_GAME_TRAIN_URL,
+  PLAY_GAME_URL,
+  USER_ME_URL,
+} from '../../shared/urls';
 import socket from '../../shared/socket';
 import { useGamePairing } from '../../shared/hooks/UseGamePairing';
 import { gameQueueClientToServerWsEvents } from 'transcendence-shared';
-import { GamePairingStatusDtoGameQueueStatusEnum } from '../../shared/generated';
+import {
+  GamePairingStatusDtoGameQueueStatusEnum,
+  User,
+} from '../../shared/generated';
+import { Link } from 'react-router-dom';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import { LandingPage } from '..';
+
+type GameWithPlayers = {
+  gameId: string;
+  playerOne: User;
+  playerTwo: User;
+};
 
 export default function PlayPage() {
   const { gameQueueStatus } = useGamePairing();
   const navigate = useNavigate();
+  const [ongoingGames, setOngoingGames] = React.useState<GameWithPlayers[]>([]);
+  const { authUser } = useAuth();
+  const [search, setSearch] = React.useState('');
 
-  // To be replaced with real games request
-  const mockFetchFn = React.useCallback(
-    (requestParams: Query): Promise<unknown[]> => {
-      return new Promise((resolve) => resolve([]));
-    },
-    [],
-  );
-  const mockedMapper = React.useCallback(
-    (data: unknown) => ({
-      iconVariant: IconVariant.ARROW_FORWARD,
-      avatarProps: {
-        url: '#',
-        XCoordinate: 0,
-        YCoordinate: 0,
-      },
-      url: '#',
-      title: '',
-      subtitle: 'level x',
-      key: '',
-    }),
-    [],
-  );
+  function gameMapper(): RowItem[] {
+    return ongoingGames.reduce<RowItem[]>((rowItems, currentGame) => {
```

Comment:
This is the same as filter and map but a bit more efficient. If it's hard to read, we could change it.